src/recipes/browserDownloadUrls.js

```diff
@@ -0,0 +1,88 @@
+/* 
+    RECIPE: browserDownloadUrls
+    -------------------------------------------------------------
+    Author: Morgan, Lia, Joel, Malick
+    Description: Looks for the download urls of other browsers
+*/
+
+
+void function() {
+    window.CSSUsage.StyleWalker.recipesToRun.push(function browserDownloadUrls(element, results){
+        //doesn't go to microsoft sites
+        if(window.location.href.toString().indexOf("microsoft.com") !== -1 || window.location.href.toString().indexOf("forum") !== -1){
+            return results;
+        }
+        function isVisible(element) {
+            //checks if width/height = 0 and left/top < 0
+            if(element.getBoundingClientRect() !== null) {
+                var box = element.getBoundingClientRect();
+                var docEl = document.documentElement;
+                var scrollTop = docEl.scrollTop;
+                var scrollLeft = docEl.scrollLeft;
+                var clientTop = docEl.clientTop;
+                var clientLeft = docEl.clientLeft;
+                var width = box.width;
+                var height = box.height;
+                var top = box.top + scrollTop - clientTop;
+                var left = box.left + scrollLeft - clientLeft;
+                var bottom = top + height;
+                var right = left + width;
+                if (width == 0 || height == 0 || bottom <= 0 || right <= 0) {
+                    return false;
+                }
+            }          
+            
+            //checks for visibility with computed style
+            var elStyle = getComputedStyle(element);
+            if(elStyle.getPropertyValue("display") === "none"){
+                return false;
+            } 
+            else if(elStyle.getPropertyValue("opacity") < 0.1){
+                return false;
+            } 
+            else if(elStyle.getPropertyValue("transform").includes(" 0,") || elStyle.getPropertyValue("transform").includes(" 0)")){
+                return false;
+            } 
+            else if(elStyle.getPropertyValue("visibility") === "hidden"){
+                return false;
+            }
+
+            // if text is within an iframe that does not appear
+            var elAbove = element;
+            do{
+                if(elAbove.nodeName === "IFRAME"){
+                    if(getComputedStyle(elAbove).getPropertyValue("src") === "" && getComputedStyle(elAbove).getPropertyValue("frameBorder") === 0){
+                        return false;
+                    }
+                }
+                if(elAbove.parentElement !== null){
```

Comment:
if not needed

      elAbove = elAbove.parentElement;
    } while(!!elAbove);